java/integTest/src/test/java/glide/SharedCommandTests.java

```diff
@@ -1011,4 +1013,92 @@ public void zcard(BaseClient client) {
+
+        String timestamp1 = "0-1";
+        assertEquals(
+                timestamp1,
+                client
+                        .xadd(
+                                key,
+                                Map.of(field1, "foo1", field2, "bar1"),
+                                StreamAddOptions.builder().id(timestamp1).build())
+                        .get());
+
+        assertNotNull(client.xadd(key, Map.of(field1, "foo2", field2, "bar2")).get());
+        if (client instanceof RedisClient) {
+            assertEquals(2L, ((RedisClient) client).customCommand(new String[] {"XLEN", key}).get());
+        } else if (client instanceof RedisClusterClient) {
+            assertEquals(
+                    2L,
+                    ((RedisClusterClient) client)
+                            .customCommand(new String[] {"XLEN", key})
+                            .get()
+                            .getSingleValue());
+        }
+
+        // this will trim the first entry.
+        String id =
+                client
+                        .xadd(
+                                key,
+                                Map.of(field1, "foo3", field2, "bar3"),
+                                StreamAddOptions.builder()
+                                        .trim(new StreamAddOptions.Maxlen(Boolean.TRUE, 2L))
+                                        .build())
+                        .get();
+        assertNotNull(id);
+        if (client instanceof RedisClient) {
+            assertEquals(2L, ((RedisClient) client).customCommand(new String[] {"XLEN", key}).get());
+        } else if (client instanceof RedisClusterClient) {
+            assertEquals(
+                    2L,
+                    ((RedisClusterClient) client)
+                            .customCommand(new String[] {"XLEN", key})
+                            .get()
+                            .getSingleValue());
+        }
+
+        // this will trim the second entry.
+        assertNotNull(
+                client
+                        .xadd(
+                                key,
+                                Map.of(field1, "foo4", field2, "bar4"),
+                                StreamAddOptions.builder()
+                                        .trim(new StreamAddOptions.MinId(Boolean.TRUE, id))
+                                        .build())
+                        .get());
+        if (client instanceof RedisClient) {
+            assertEquals(2L, ((RedisClient) client).customCommand(new String[] {"XLEN", key}).get());
+        } else if (client instanceof RedisClusterClient) {
+            assertEquals(
+                    2L,
+                    ((RedisClusterClient) client)
+                            .customCommand(new String[] {"XLEN", key})
+                            .get()
+                            .getSingleValue());
+        }
+
+        /**
+         * TODO add test to XTRIM on maxlen expect( await client.xtrim(key, { method: "maxlen",
```

Comment:
You don't need the todo you can just add it later when you add the Xtrim command.